dragonfly config stuff: rack_cache=true, ragonfly_accessor :image, rack-...

Dana-G opened the pull request:
dragonfly config stuff: 
rack_cache=true was uncommented, 
dragonfly_accessor :image added to user model b/c I have an avatar in that model...correct?
rack-cache gem added to new :production group
(fyi- rspec is already in there from first commit)

mcelaney commented on config/routes.rb:
```diff
@@ -1,11 +1,13 @@
 Rails.application.routes.draw do
   get 'users/new'
 
+  match '/media/:dragonfly/:file_name', :to => Dragonfly[:images]
```
Prefer not to use hash rockets.
match '/media/:dragonfly/:file_name', to: Dragonfly[:images]

Dana-G commented:
Also removed the +> from routes file. 
added rubocop, passed tests, 
added rubocop output file, 
migrated new table authenticaton columns (removed passord, password_hash and password_salt) (added password_digest)

Dana-G commented:
expanded user controller to allow user edits, and signin. Added stron…
…g user parameters, Created affiliated usr views. Added haml-rails gem. Added static home page

Dana-G commented:
in the middle of railscast 274 - added remember me functionality. Add…
…ed controller and mailer for reset password (not working yet)

mcelaney commented on .rubocop_todo.yml:
```diff
@@ -0,0 +1,84 @@
+# This configuration was generated by `rubocop --auto-gen-config`
```
I'd just remove this file.  It's intended to help make knocking a large codebase with a lot of offenses easier.

mcelaney commented on app/controllers/application_controller.rb:
```diff
@@ -1,5 +1,21 @@
+private
+
+  def current_user
+    @current_user ||= User.find_by_auth_token!(cookies[:auth_token]) if cookies[:auth_token]
```
Line too long

mcelaney commented on app/controllers/application_controller.rb:
```diff
@@ -1,5 +1,21 @@
+  helper_method :current_user, :logged_in?
+
+  def authorize
+    redirect_to login_url, alert: "Not authorized" if current_user.nil?
```
Run Rubocop (single quotes unless stuff)

mcelaney commented on app/controllers/password_resets_controller.rb:
```diff
@@ -0,0 +1,10 @@
+
+  def create
+    user = User.find_by_email(params[:email])
+    user.send_password_reset if user
```
Ask about the null object pattern for fun and profit.

Dana-G commented:
finished railscasts 247, updated tutorial info to current, fixed spec…
… errors due to version, started running basic tests - currently working, reinstalled spec_helper, rails_helper, rspec files - renamed and kept original

and also
	ran rubocop, almost all guides successfully followed
(did not yet address earlier commit comments)
